lighthouse-core/scripts/lantern/minify-trace.js

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env node
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-disable no-console */
+
+const fs = require('fs');
+const path = require('path');
+
+if (process.argv.length !== 4) {
+  console.error('Usage $0: <input file> <output file>');
+  process.exit(1);
+}
+
+const inputTracePath = path.resolve(process.cwd(), process.argv[2]);
+const outputTracePath = path.resolve(process.cwd(), process.argv[3]);
+const inputTraceRaw = fs.readFileSync(inputTracePath, 'utf8');
+/** @type {LH.Trace} */
+const inputTrace = JSON.parse(inputTraceRaw);
+
+const toplevelTaskNames = new Set([
+  'TaskQueueManager::ProcessTaskFromWorkQueue',
+  'ThreadControllerImpl::DoWork',
+]);
+
+const traceEventsToAlwaysKeep = new Set([
+  'Screenshot',
+  'TracingStartedInBrowser',
+  'TracingStartedInPage',
+  'navigationStart',
+  'ParseAuthorStyleSheet',
+  'ParseHTML',
+  'PlatformResourceSendRequest',
+  'ResourceSendRequest',
+  'ResourceReceiveResponse',
+  'ResourceFinish',
+  'ResourceReceivedData',
+  'EventDispatch',
+]);
+
+const traceEventsToKeepInToplevelTask = new Set([
+  'TimerInstall',
+  'TimerFire',
+  'InvalidateLayout',
+  'ScheduleStyleRecalculation',
+  'EvaluateScript',
+  'XHRReadyStateChange',
+  'FunctionCall',
+  'v8.compile',
+]);
+
+const traceEventsToKeepInProcess = new Set([
+  ...toplevelTaskNames,
+  ...traceEventsToKeepInToplevelTask,
+  'firstPaint',
+  'firstContentfulPaint',
+  'firstMeaningfulPaint',
+  'firstMeaningfulPaintCandidate',
+  'loadEventEnd',
+  'domContentLoadedEventEnd',
+]);
+
+/**
+ * @param {LH.TraceEvent[]} events
+ */
+function filterTraceEvents(events) {
+  const startedInPageEvt = events.find(evt => evt.name === 'TracingStartedInPage');
```

Comment:
lucky you: https://github.com/GoogleChrome/lighthouse/pull/5271 :)

Reply:
speaking of which, is there some way to share all this logic with `trace-of-tab` so things like this (and what's considered "important") are always kept in sync?

Reply:
yeah I'd want #5271 to move the logic into tracing-process for mainthread identification and then I can just reuse it :D

Reply:
ok, so...future TODO/file an issue/wait until something breaks? :)

Reply:
TODO it is!